### PR TITLE
introduce prop-types instead of React.PropTypes

### PR DESCRIPTION
**Pull request by nekoya**
#### Pull request checklist

- [x] Addresses an existing issue: Relates #1740
- [ ] Include a change request file if publishing 
- [x] New feature, bugfix, or enhancement
  - [ ] Includes tests
- [ ] Documentation update

#### Description of changes

At React v15.5, `React.PropTypes` and `react-addons-test-utils` are deprecated.

We depends React v15.4 still, I feel that it is hard to update React v15.5.

This PR resolve the former warnings (about `React.PropTypes`).

This update fixes the warning from React v15.5 on users' application.

#### Additional question

I caught some errors when I build it on local machine, through `npm install` and `npm test` like below.

```
Error - typescript - src/Customizer.tsx(1,27): error TS2307: Cannot find module 'prop-types'.
Error - typescript - src/customizable.tsx(1,27): error TS2307: Cannot find module 'prop-types'.
Error - 'typescript' sub task errored after 1.85 s
```

In my understanding `packages/*/node_modules` are symlinks to `common/node_modules`, so I registered `prop-types` to `common/package.json` and `common/pinnedVersions.json` (this PR is not contain such changes).

However I couldn't fix these errors. How can I fix it correctly?

### CODE REVIEW

**Comment by micahgodbolt** [comment]
Great job. I saw this deprecation in another project recently and didn't even realize we were using React.PropTypes in our project. I'd probably flag down @dzearing about the build issues.

File: packages/office-ui-fabric-react/package.json [hunk 1]
```diff
@@ -47,6 +48,7 @@
     "node-sass": "^4.5.0",
     "office-ui-fabric-core": ">=5.0.0 <6.0.0",
     "postcss-loader": "^1.3.3",
+    "prop-types": "^15.5.9",
```
Comment (dzearing) [inline packages/office-ui-fabric-react/package.json]: You are adding the `prop-types` package as a dev dependency but referring to it in production code. It is a dependency, not a dev dependency.

File: packages/office-ui-fabric-react/package.json [hunk 2]
```diff
@@ -22,6 +22,7 @@
     "@types/enzyme": "^2.7.5",
```
Comment (dzearing) [inline packages/office-ui-fabric-react/package.json]: Please use `rush change` to add a change file to patch fabric-react and add a nice comment describing this, which will show up in the changelog.

File: packages/office-ui-fabric-react/package.json (hunk 1 above)
Comment (nekoya) [inline packages/office-ui-fabric-react/package.json]: Is that really? If so, why `react` is registered into devDependencies?

I only replaced `React.PropType` to `PropTypes`, I think its dependency was not changed essentially.

File: packages/office-ui-fabric-react/package.json (hunk 2 above)
Comment (nekoya) [inline packages/office-ui-fabric-react/package.json]: I'm not familiar with `rush` and I have to study it. Thank you for suggestion.

File: packages/office-ui-fabric-react/package.json (hunk 1 above)
Comment (cschleiden) [inline packages/office-ui-fabric-react/package.json]: Let's not confuse types and dependencies. `react` itself is registered as a `peerDependency`. The guidance for `prop-types` https://www.npmjs.com/package/prop-types recommends adding `prop-types` as a dependency.

**Comment by cschleiden** [comment]
Just to be sure, will this change play nice with projects that use office fabric but still use a pre 15.5 version of rect?

File: packages/utilities/package.json [hunk 3]
```diff
@@ -40,4 +42,4 @@
     "react": "^0.14 || ^15.0.1-0 || ^16.0.0-0",
```
Comment (cschleiden) [inline packages/utilities/package.json]: Do we have to upgrade to 0.14.9 now? 

https://github.com/facebook/prop-types#compatibility

File: packages/office-ui-fabric-react/package.json (hunk 2 above)
Comment (dzearing) [inline packages/office-ui-fabric-react/package.json]: It's easy. 

```base
npm i -g @microsoft/rush
rush change -b upstream/master 
```

Then answer questions. "upstream/master" refers to the OfficeDev remote master branch, you may need to register "upstream" pointing to the OfficeDev remote.

File: packages/office-ui-fabric-react/package.json (hunk 1 above)
Comment (nekoya) [inline packages/office-ui-fabric-react/package.json]: Certainly. I'll fix it.

File: packages/utilities/package.json (hunk 3 above)
Comment (nekoya) [inline packages/utilities/package.json]: I've confirmed the following warning message when I build an application with react/react-dom 0.14.8 and prop-types 15.5.10.

>warning.js:45 Warning: Failed propType: Calling PropTypes validators directly is not supported by the `prop-types` package. Use `PropTypes.checkPropTypes()` to call them. Read more at http://fb.me/use-check-prop-types Check the render method of `AccountingView`.

React v0.14.9 works fine, without such a warning. Therefore I think we have to upgrade to 0.14.9.

Meanwhile, I have a concern too. React v0.14.9 does not appeared on its [CHANGELOG](https://github.com/facebook/react/blob/master/CHANGELOG.md). It seems a simple [mistake](https://twitter.com/dan_abramov/status/852193956322521089), but I suspect React v0.14.9 is not famous yet.

On the other hand, React v0.14.9 is a valid release version, we can install it by `^0.14.0`.

In conclusion, we should fix React version like below.

```
"peerDependencies": {
  "react": "^0.14.9 || ^15.0.1-0 || ^16.0.0-0",
  "react-dom": "^0.14.9 || ^15.0.1-0 || ^16.0.0-0"
}
```

How do you feel about it?

File: packages/office-ui-fabric-react/package.json (hunk 2 above)
Comment (dzearing) [inline packages/office-ui-fabric-react/package.json]: Woops. Recently there was an update to rush, which is not backwards compatible and may not work.

Instead, you can use the locally installed copy by running this from the root:

```bash
npm run change
```

File: packages/utilities/package.json (hunk 3 above)
Comment (dzearing) [inline packages/utilities/package.json]: i am fine with that.

**Comment by nekoya** [comment]
Sorry for late, I've fixed dependencies, and I succeeded to pass tests (the directory named `common/temp_modules` made me confused).

I'd like to squash these commits before merging.

**Comment by nekoya** [comment]
I resolved conflict and squashed commits (as a result, The commit by @dzearing is squashed too, sorry).

**Comment by stephenjelfs** [comment]
I'm still getting warnings with the latest version of office-ui-fabric-react, 2.34.2 due to the following dependency:

"@uifabric/utilities": ">=1.10.2 <2.0.0"

Maybe the version of this package also needs to bumped?

**Comment by nekoya** [comment]
This PR is released at 4.0.0, not 2.34.2. And I've confirmed that the warning is fixed.

**Comment by stephenjelfs** [comment]
@nekoya Great, thanks. I also checked this with React 16.alpha.12 and this works too.
